nav2_smac_planner/src/smac_planner_lattice.cpp

```diff
@@ -83,6 +83,11 @@ void SmacPlannerLattice::configure(
   nav2_util::declare_parameter_if_not_declared(
     node, name + ".cache_obstacle_heuristic", rclcpp::ParameterValue(false));
   node->get_parameter(name + ".cache_obstacle_heuristic", _search_info.cache_obstacle_heuristic);
+  nav2_util::declare_parameter_if_not_declared(
```

Comment:
Actually I don't see a reason for keeping the old implementation, I left the option there just because you were quite resistive against the change in our [discussion](https://github.com/ros-planning/navigation2/issues/2498) because of computation speed. Our tests have, however, shown that the new heuristic computation is in general faster (and more importantly it yields correct results) but there are some theoretical situations where the old can be faster so I'm leaving the decision about deleting the old upon you.

If needed for benchmarking, you can check out our [smac_planner_debug](https://github.com/robotechvision/navigation2/tree/smac_planner_debug) branch which contains some heuristic computation time logs, but maybe you have your own tools.

Reply:
Got it, do you have some relative numbers to share on the compute time saved (on average)? It's obviously map / location dependent, but some ballparks to start would be nice. I can then use that to see if I can get similar numbers in my maps to verify.

If its not too much more compute, or even better yet, _less_  compute, then I'd be happy to have my implementation striped out completely. Lets leave in for now so I can do that benchmarking more easily side-by-side. Let me know on the metrics you see.

Reply:
I have tested it some time ago and for an 80m long path from one map corner to another both computations lasted around 40 ms. 4x speedup came with a path (25m: 16ms vs 4ms) leading to a goal which is in the middle of the map since the inadmissible implementation searches in every direction from goal while the new 2D A* implementation searches from goal mostly towards starting node

Reply:
Got it -- I think then the only thing remaining in this line item is the `This is missing from the readme list of variables.`